cirrus/google_cloud/manager.py

```diff
@@ -1124,6 +1127,22 @@ def remove_member_from_group(self, member_email, group_id):
             if err.resp.status == 404:
                 # not found, member isn't in group. This is fine
                 response = {}
+            elif err.resp.status == 400:
+                # Google's API erroneously returns 400 sometimes
+                # we check to see if the SA was actually deleted
+                logger.warning(
+                    "When removing {} from group ({}), Google API "
+                    "returned status 400".format(member_email, group_id)
+                )
+                member_emails = [
+                    member.get("email", "")
+                    for member in self.get_group_members(group_id)
+                ]
+                if member_email in member_emails:
+                    raise
+                else:
+                    # reaching this point, indicates the member was succesfully removed
+                    response = {}
```

Comment:
should log info saying after 400 you got group members from google's api and were able to verify the account was removed